### PR TITLE
Registry for OID/Evidence Mappings

### PR DESCRIPTION
**Pull request by hannestschofenig**

### CODE REVIEW

File: draft-ietf-lamps-csr-attestation.md [hunk 1]
```diff
@@ -631,6 +628,46 @@ Columns:
...
+
+~~~
+ +-----------------------------------------------------------------------+
+ | OID              | Description     | Reference(s) | Change Controller |
+ +------------------+-----------------+--------------+-------------------+
+ | 2 23 133 5 4 10  | DICE Evidence   | [TCGDICE1.1] |  TCG              |
+ | 2 23 133 5 4 9   | Conceptual      | [TCGDICE1.1] |  TCG              |
+ |                  | Message Wrapper |              |                   |
+ +------------------+-----------------+--------------+-------------------+
+~~~
```
Comment (thomas-fossati) [inline draft-ietf-lamps-csr-attestation.md]: use kramdown-rfc table syntax; also, make reference to TCG a real ref

```suggestion
| OID              | Description                | Reference(s)   | Change Controller |
|------------------|----------------------------|----------------|-------------------|
| 2 23 133 5 4 10  | DICE Evidence              | {{TCGDICE1.1}} |  TCG              |
| 2 23 133 5 4 9   | Conceptual Message Wrapper | {{TCGDICE1.1}} |  TCG              |
{: #tab-ae-reg title="Initial Contents of the Attestation Evidence OID Registry"}
```

File: draft-ietf-lamps-csr-attestation.md [hunk 2]
```diff
@@ -934,7 +971,7 @@ information to an RA/CA:
 
 ## TCG DICE ConceptualMessageWrapper in CSR
 
-This section gives an example of extending the ASN.1 module above to carry an existing ASN.1-based evidence statement. The example used is the Trusted Computing Group DICE Attestation Conceptual Message Wrapper as defined in [TCGDICE1.1].
+This section gives an example of extending the ASN.1 module above to carry an existing ASN.1-based evidence statement. The example used is the Trusted Computing Group DICE Attestation Conceptual Message Wrapper, as defined in [TCGDICE1.1].
```
Comment (thomas-fossati) [inline draft-ietf-lamps-csr-attestation.md:988]:
```suggestion
This section gives an example of extending the ASN.1 module above to carry an existing ASN.1-based evidence statement. The example used is the Trusted Computing Group DICE Attestation Conceptual Message Wrapper, as defined in {{TCGDICE1.1}}.
```

**Review by thomas-fossati — APPROVED** [review]
This seems like a very useful addition.

I am wondering whether the new registry's scope could be extended to cover all typing information related to attestation evidence, including OID, media types, CoAP C-F, CBOR tags, etc., to make this a one-stop shop "meta registry" for everything attestation.
Not sure if there is a precedence for this in IANA.


PS: I have left a couple of markdown-related fixes in-line.

File: draft-ietf-lamps-csr-attestation.md [hunk 3]
```diff
@@ -631,6 +628,60 @@ Columns:
 -  Description: Begins with id-ata
 -  References: RFC or other document
 
+## Attestation Evidence OID Registry
+
+IANA is asked to create a registry that helps developers to find
```
Comment (thomas-fossati) [inline draft-ietf-lamps-csr-attestation.md:633]: fix lint error (and clear the CI)

```suggestion
IANA is asked to create a registry that helps developers to find
```

File: draft-ietf-lamps-csr-attestation.md [hunk 4]
```diff
@@ -631,6 +629,58 @@ Columns:
...
+
+### Registration Template
+
+The registry has the following columns:
+
+- OID: The number of OID.
+
+- Description: Brief description of the use of the Evidence and the
+registration of the OID.
+
```
Comment (thomas-fossati) [inline draft-ietf-lamps-csr-attestation.md:660]: more fix-lint :-)

```suggestion

```

File: draft-ietf-lamps-csr-attestation.md [hunk 5]
```diff
@@ -590,30 +591,27 @@ TypedFlatCert ::= SEQUENCE {
...
+##  Module Registration - SMI Security for PKIX Module Identifier
 
 -  Decimal: IANA Assigned - **Replace TBDMOD**
 -  Description: CSR-ATTESTATION-2023 - id-mod-pkix-attest-01
 -  References: This Document
 
-###  Object Identifier Registrations - SMI Security for S/MIME Attributes
+##  Object Identifier Registrations - SMI Security for S/MIME Attributes
 
 - Attest Statement
```
Comment (henkbirkholz) [inline draft-ietf-lamps-csr-attestation.md:607]:
```suggestion
- Evidence Statement
```
I know, I know... but it would make more sense, right?

Comment (henkbirkholz) [reply draft-ietf-lamps-csr-attestation.md:607]: Only occurrence. But if out-of-scope here, this can become a tiny new issue.

**Review by henkbirkholz — APPROVED** [review]

**Comment by ounsworth** [comment]
Meeting 2024-01-29
This PR is mergeable, pending sorting out the [[TBD]] mail list reference.

**Comment by ounsworth** [comment]
As per discussion on the RATS mailing list, this is considered a very good idea, and is slipping into scope creep.

At this point, I suggest removing this from the CSR Attestation draft so that the CSR draft is not delayed due to this much larger initiative.

(aka, the CA industry really needs this CSR draft ASAP. Can we please put a New Content Freeze on it?)

https://mailarchive.ietf.org/arch/msg/rats/h98spAElEuQtpIEWStEuaCyRQrI/

**Comment by henkbirkholz** [comment]
> I am wondering whether the new registry's scope could be extended to cover all typing information related to attestation evidence, including OID, media types, CoAP C-F, CBOR tags, etc., to make this a one-stop shop "meta registry" for everything attestation.
Not sure if there is a precedence for this in IANA.

@thomas-fossati, could you imagine for this endeavor to move into a separate document?

**Comment by thomas-fossati** [comment]
Sorry, replied to Mike's email before reading this.

> @thomas-fossati, could you imagine for this endeavor to move into a separate document?

Yes, absolutely.

**Comment by henkbirkholz** [comment]
then we can :ship: this ☺️

File: draft-ietf-lamps-csr-attestation.md [hunk 6]
```diff
@@ -631,6 +629,63 @@ Columns:
...
 
+## Attestation Evidence OID Registry
+
+IANA is asked to create a registry that helps developers to find
+OID/Evidence mappings.
+
+Registration requests are evaluated using the criteria described in
+the registration template below after a three-week review period on
+the [[TBD]] mailing list, on the advice of one or more Designated
+Experts {{RFC8126}}.  However, to allow for the allocation of values
```
Comment (mcr) [inline draft-ietf-lamps-csr-attestation.md:640]:
```suggestion
the registration template below after a three-week review period on
the [[TBD]] mailing list, with the advice of one or more Designated
Experts {{RFC8126}}.  However, to allow for the allocation of values
```

I think that it's a three-week review,, *AND* advice of DEs.  With the previous wording, I could interpret "or"

**Review by mcr — CHANGES_REQUESTED** [review]
As I understand it, we are not asking for a new OID arc, but rather just collecting the OIDs that are used from other arcs?

**Review by ounsworth — APPROVED** [review]

**Pull request merged**